### PR TITLE
feat(A11Y): improve a11y on toc component

### PR DESCRIPTION
**Pull request by dianjuar**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/nestjs/docs.nestjs.com/blob/master/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Currently the toc component has two problems related with a11y (accessibility):
- Is not focusable
- Does not indicate where you are going to go (you can not the preview URL)

## What is the new behavior?
The new improvements are:
- The toc elements are focusable
- You can see where you are going to go (you can see the URL)
- When you select a title, the focus is transferred to the title to continue browsing
- The focus flow is changed
  - image
  - Now you pass from the navbar (N° 2) to the toc component (N° 3), because has no sense to pass through all the article to go there.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/styles.scss [hunk 1]
```diff
@@ -4,6 +4,10 @@
 @import './scss/variables.scss';
 @import './scss/utils.scss';
 
+:root {
+  scroll-behavior: smooth;
```
Comment (kamilmysliwiec) [inline src/styles.scss]: Isn't it too early to use this property?

Comment (dianjuar) [reply src/styles.scss]: Because is not fully [implemented](https://caniuse.com/#search=scroll-behavior) ?

Comment (dianjuar) [reply src/styles.scss]: Quoting the css tricks article [_Smooth Scrolling and Accessibility_](https://css-tricks.com/smooth-scrolling-accessibility/#article-header-id-0)

>  when you follow a link, the keyboard focus should follow it, too and be able to access the next element after the target. 

Here is an example

### [demo](https://output.jsbin.com/yilokec)
example

#### To implement a smooth scrolling with that recommendation the best way that I could find is with that css property. 

----------------

## Why?
The current implementation of the smooth scrolling do not update the keyboard focus.
```js
elementRef.scrollIntoView({	
  behavior: 'smooth',	
  block: 'start'	
});
```

There is a lack of callback to know when the scroll finish to update the focus to the new element and if you do it right away like this
```js
elementRef.scrollIntoView({	
  behavior: 'smooth',	
  block: 'start'	
});
elementRef.focus();
```
When you `elementRef.focus()`, the target hijacks the smooth scrolling and it jumps right at the target, abruptly.

----------------

The possible solutions that I found were:
- Use Jquery animate
- Implement a `setTimeout` with the same duration as the scrolling
- Use the css property `:root { scroll-behavior: smooth; }`

I don't want to add a library like jquery just for that, making the web page load slow.
Agreeing with a [comment](https://css-tricks.com/smooth-scrolling-accessibility/#comment-1608095) in the article, use a `setTimeout` doesn't feel particularly elegant.

If you look at [Can I Use](https://caniuse.com/#search=scroll-behavior) the porcentaje of usage is 71.42% considering all users (desktop and mobile), doesn't look too bad like the other options.

image

Comment (dianjuar) [reply src/styles.scss]: There is something that is not clear o I need to go deeper ?

Comment (kamilmysliwiec) [reply src/styles.scss]: The idea is great. I'm just afraid about browser compatibility for now. 71% isn't that much

Comment (dianjuar) [reply src/styles.scss]: The 29% left, just is not going to have a smooth scroll, the functionality is going to be the same. Their jump is going to be immediate

Comment (dianjuar) [reply src/styles.scss]: You can make the test in an old browser (chrome 59) that doesn't support the css property

https://www.browserling.com/browse/win/7/chrome/59/https%3A%2F%2Fdeploy-preview-433--docs-nestjs.netlify.com%2F

Comment (dianjuar) [reply src/styles.scss]: It rises up to [75.97%](https://caniuse.com/#search=scroll-behavior)

**Comment by kamilmysliwiec** [comment]
Thanks!

**Pull request merged**